cmd/end2endtest/helpers.go

```diff
@@ -183,123 +181,65 @@ func (t *e2eElection) waitUntilElectionStarts(electionID types.HexBytes) (*vapi.
 	return election, nil
 }
 
-func (t *e2eElection) generateSIKProofs(root types.HexBytes) map[string]*apiclient.CensusProof {
-	type voterProof struct {
-		sikproof *apiclient.CensusProof
-		address  string
-	}
-	sikProofs := make(map[string]*apiclient.CensusProof, len(t.voterAccounts))
-	proofCh := make(chan *voterProof)
-	stopProofs := make(chan bool)
-	go func() {
-		for {
-			select {
-			case p := <-proofCh:
-				sikProofs[p.address] = p.sikproof
-			case <-stopProofs:
-				return
-			}
-		}
-	}()
-
+func (t *e2eElection) generateProofs(csp *ethereum.SignKeys, voterAccts []*ethereum.SignKeys) error {
+	var (
+		wg     sync.WaitGroup
+		vcount int32
+	)
+	errorChan := make(chan error, 1)
+	t.voters = new(sync.Map)
 	addNaccounts := func(accounts []*ethereum.SignKeys, wg *sync.WaitGroup) {
```

Comment:
no need to pass `wg` as an argument anymore, since you moved the `var wg     sync.WaitGroup` to the top

i.e. just do
```
addNaccounts := func(accounts []*ethereum.SignKeys) {
```

Reply:
Fixed, thanks!